src/sagas/walletConnect.js

```diff
@@ -517,6 +524,45 @@ export function* onSessionDelete(action) {
   yield call(onCancelSession, action);
 }
 
+/**
+ * Request tokens data to feed walletConnect's tokens.
+ * @param {Object} action
+ * @param {Object} action.payload
+ * @param {string[]} action.payload.uids
+ */
+export function* requestTokens(action) {
+  const { uids } = action.payload;
+
+  const wallet = yield select((state) => state.wallet);
+  if (!wallet.isReady()) {
+    log.error('Fail updating loading tokens data because wallet is not ready yet.');
+    // This will show user an error modal with the option to send the error to sentry.
+    yield put(onExceptionCaptured(new Error(failureMessage.walletNotReadyError), true));
+    return;
+  }
+
+  const tokens = {};
+  let someError = false;
+  for (const uid of uids) {
+    try {
+      const { tokenInfo: { symbol, name } } = yield call([wallet, wallet.getTokenDetails], uid);
+      const token = { uid, symbol, name };
+      tokens[uid] = token;
+    } catch (e) {
+      log.error(`Fail getting token data for token ${uid}.`, e);
+      someError = true;
+      // continue
+    }
+  }
```

Comment:
You can use `all` to make the requests in parallel, also here goes a suggestion for legibility:

```suggestion
const tokenDetailsRequests = uids.map((uid) =>
  call(function* () {
    try {
      const { tokenInfo: { symbol, name } } = yield call([wallet, wallet.getTokenDetails], uid);

      return [false, { uid, symbol, name }];
    } catch (e) {
      log.error(`Fail getting token data for token ${uid}.`, e);
      someError = true;
      return [true, null];
    }
  })
);

const [someError, tokenDetails] = (yield all(tokenDetailsRequests)).reduce(
  (acc, tokenDetailRequest) => {
    const [error, tokenObj] = tokenDetailRequest;
    const [lastError, tokenDetails] = acc;

    return [
      lastError || error,
      tokenDetail ? {
        ...tokenDetails,
        [tokenDetail.uid]: tokenDetail
      } : tokenObj,
    ];
  }, [false, {}]
);
```

Also, can you move this code to `src/sagas/tokens.js` and call it here?

Reply:
Oh, thank you! Tbh, I have to think more about it. It involves network calls and may have some hidden pitfalls. I have to test it first. I have had some problems with "parallelism" using promises before. Lek me do some tests first. I'm opening this issue #531 to work on it later.

Reply:
I don't think you'll have any problems here, even with a huge number of tokens

React-native's native HTTP module will queue them for you and only resolve when the request is done

But I agree that this should be tested, so OK to doing it in another PR